modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/components/chart/bar/SimpleBarChart.es.js

```diff
@@ -49,9 +53,13 @@ export default ({data, height, totalEntries, width}) => {
 							? 'dim'
 							: ''
 					}`}
-					x={-22}
+					x={minX}
 				>
-					{`${payload.value}`}
+					{`${
+						payload.value.replace(/\s/g, '').length > maxTextSize
```

Comment:
@rebsilva, is it really necessary replace blank spaces before check the text length? Why?

If it's necessary, can you please add the regex `\s` to a dedicated variable to improve code legibility like in this [code](https://github.com/liferay/liferay-portal/blob/7618f439fe4945b1574d16a87de7c79680d361ee/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/util/strings.es.js#L19)?

Reply:
Hi Renato! In this case is necessary, I will try to explain you why:

When a user writes an option with a lot of spaces like this `He l   l      o!` it will show on the Forms and Charts like this `He l l o!` (it doesn't count consecutive white spaces). But when we use the `payload.value` it will return us the original `He l   l      o!` with a lot of white spaces; if we use `payload.value.length` directly it will count all the white spaces and a small word like `He l l o!` will suffer the ellipsize which is something we don't want. This way, we use the regex `/\s/g` to replace all the white spaces of a string (e.g. `He l   l      o!` will be `Hello!`) and count properly the length of the option. If you have any more questions about this or it wasn't a clear explanation please let me know!

I will also put the regex on a variable, thanks for the heads up!

Reply:
Thanks for your detailed explanation, @rebsilva. :slightly_smiling_face:

Reply:
No problem!